app.ts

```diff
@@ -0,0 +1,147 @@
+            const th = document.createElement('th');
+            this.myTable.appendChild(th);
+
+            th.appendChild(document.createTextNode(" " + i));
+        }
+
+        // write html for table
+        for (const iEntry of recordsTable) {
+
+            const tr = document.createElement('tr');
+            this.myTable.appendChild(tr);
+
+            for (const jEntry of iEntry) {
+
+                const td = document.createElement('td');
+                td.appendChild(document.createTextNode(" " + jEntry));
+                tr.appendChild(td);
+            }
+        }
+    }
+}
+
+function createPage(firstRecord: number) {
+
+    // ensure that the user input is a number and within range of available records
+    if (firstRecord >= RECORD_COUNT || firstRecord < 0 || isNaN(firstRecord)) {
+        // if (firstRecord >= RECORD_COUNT || firstRecord < 0) {
+        alert('please select a NUMBER between 0 and ' + (RECORD_COUNT - 1));
+        return;
+    }
+
+    const contentsDiv = this.content;
+    contentsDiv.innerHTML = ''; // empty table div to fill in a new table
+    const tableElement = new tableGenerator(contentsDiv, firstRecord); // create table using the tableGenerator class
+    tableElement.generateTable();
+    LAST_ENTRY = tableElement.endRecord;
+
+    // keep the ID of the first and last table entries to use for next and previous page buttons
+    // LAST_ENTRY = contentsDiv.lastChild.lastChild.firstChild.textContent;
+    CURRENT_ENTRY = firstRecord;
+};
+
+// fucntion used by the next page buttton, requires the last entry from previously generated table
+function nextPage() {
+    createPage(parseInt(LAST_ENTRY) + 1);
+}
+
+// fucntion used by the previous page buttton, requires the first and last entry from previously generated table
+function prevPage() {
+
+    const newfirstEntry = CURRENT_ENTRY - (LAST_ENTRY - CURRENT_ENTRY) - 1;
+
+    // if the new first entry is less than zero, take user to the first page 
+    newfirstEntry <= 0 ? createPage(0) : createPage(newfirstEntry);
+}
+
+// hint for the user
+function helpFunc() {
+    alert(`
+    Table for viewing records with IDs 0 to ${(RECORD_COUNT - 1)} (Numbers Only). 
+    To start from a specific record enter a record ID and click "Go To".`)
+}
+
+window.onresize = () => {
+    const doneResizing = () => {
+        createPage(CURRENT_ENTRY);
+    }
+
+    clearTimeout(resizeTimer);
+    resizeTimer = setTimeout(doneResizing, 500);
+}
+
+// load page for the first time and create buttons
+window.onload = () => {
+
+    const controlsDiv = this.controls;
+    CONTROLS_HEIGHT = controlsDiv.offsetHeight;
+
+    // html for buttons and input text fields
+    controlsDiv.innerHTML += `
+    <button onclick = "createPage(0)">First Page</button>
+    <button onclick = "prevPage()" > Prev </button>
+    <button onclick = "nextPage()" > Next </button>
+    <button onclick = "createPage(document.getElementById(&quot;recordNum&quot;).value)"> Go To </button>
```

Comment:
Storing program state in the UI (DOM) is a no-no. Let's find a way around it.

Reply:
Actually I see you're not really storing it in DOM, but prettier to have all your code in JS, not in HTML. You have all the advantages of code-completion and type checking in the JS, none in the HTML.